internal/consul/namespaces.go

```diff
@@ -0,0 +1,50 @@
+package consul
+
+import (
+	capi "github.com/hashicorp/consul/api"
+)
+
+const (
+	WildcardNamespace = "*"
+	DefaultNamespace  = "default"
+)
+
+// EnsureNamespaceExists ensures a Consul namespace with name ns exists. If it doesn't,
+// it will create it and set crossNSACLPolicy as a policy default.
+// Boolean return value indicates if the namespace was created by this call.
+func EnsureNamespaceExists(client *capi.Client, ns string, crossNSAClPolicy string) (bool, error) {
```

Comment:
```suggestion
func EnsureNamespaceExists(client *capi.Client, ns string, crossNSACLPolicy string) (bool, error) {
```

Reply:
It seems like `crossNSACLPolicy` is only ever `""` in our codebase. Do we need to support it anyways, or could we drop the extra parameter for now and add back later if needed?

Similarly, we never actually use the `bool` return value. Could we simplify this function to just return the `error`?

Reply:
I'm ok with dropping crossNSACLPolicy and its logic since I don't think we really want our controller to be in character of NSACL policy to begin with, but I'll keep the boolean for now.